aslam_offline_calibration/kalibr/python/kalibr_visualize_distortion

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+import argparse
+import kalibr_common as kc
+import math
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import numpy as np
+import sm
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description=
+        'Visualizes a camchain-cam.yaml distortion.')
+    parser.add_argument(
+        '--cam',
+        dest='chain_yaml',
+        help=
+        'Camera configuration as yaml file (can take multiple configurations).',
+        nargs='+',
+        required=True)
+    parser.add_argument(
+        '--number_of_rectangles',
+        help="Number of rectangles to draw.",
+        default=5)
+    parser.add_argument(
+        '--line_thickness',
+        help="Line thickness.",
+        default=3)
+    parsed_args = parser.parse_args()
+    return parsed_args
+
+def normalize((u,v), focal_lengths, principal_point):
+    x = (u - principal_point[1]) / focal_lengths[1]
+    y = (v - principal_point[0]) / focal_lengths[0]
+    return (x,y)
+
+def denormalize((x,y), focal_lengths, principal_point):
+    u = x * focal_lengths[1] + principal_point[1]
+    v = y * focal_lengths[0] + principal_point[0]
+    return (u,v)
+
+def distortPoint((x,y), distortion_model, distortion_coeffs):
+    if distortion_model == "radial-tangential" or distortion_model == "radtan":
+        x2 = x**2
+        y2 = y**2
+        xy = x * y
+        rho2 = x2 + y2
+        rad_dist = distortion_coeffs[0] * rho2 + distortion_coeffs[1] * rho2**2
+        x += x * rad_dist + 2.0 * distortion_coeffs[2] * xy + \
+            distortion_coeffs[3] * (rho2 + 2.0 * x2)
+        y += y * rad_dist + 2.0 * distortion_coeffs[2] * xy + \
+            distortion_coeffs[3] * (rho2 + 2.0 * y2)
+        return (x, y) 
+    elif distortion_model == "equidistant" or distortion_model == "equi":
+        r = math.sqrt(x**2 + y**2)
+        if r < 1e-8:
+          return (x,y)
+        theta = math.atan(r)
+        thetad = theta * (1 + distortion_coeffs[0] * theta**2  + \
+            distortion_coeffs[1] * theta**4 + distortion_coeffs[2] * \
+            theta**6 + distortion_coeffs[3] * theta**8)
+        scaling = thetad / r
+        return (scaling * x, scaling * y)
+    elif distortion_model ==  "fov":
+        w = distortion_coeffs[0]
+        r_u = np.linalg.norm((x,y))
+        tanwhalf = math.tan(w / 2)
+        atan_wrd = math.atan(2. * tanwhalf * r_u)
+        
+        r_rd = 0.0
+        if w * w < 1e-5:
+          r_rd = 1.0
+        else:
+          if r_u * r_u < 1e-5:
+            r_rd = 2. * tanwhalf / w
+          else:
+            r_rd = atan_wrd / (r_u * w)
+        return (x * r_rd, y * r_rd)
+    else:
+      raise Exception('Distortion model ' + distortion_model + ' not found.')
+
+def distort(image, distortion_model, distortion_coeffs, focal_lengths, 
```

Comment:
probably it would be faster to use the aslam_cv_python implementation of the distort function, but on the other hand it doesn't matter so much here..

Reply:
It probably would be. But I was too lazy to figure out the interface ;-)